src/sysupdate/sysupdated.c

```diff
@@ -1131,11 +1131,15 @@ static int target_method_vacuum_finish(
                 sd_json_variant *json,
                 sd_bus_error *error) {
 
+        sd_json_variant *v;
         uint64_t instances;
 
         assert(json);
 
-        instances = sd_json_variant_unsigned(sd_json_variant_by_key(json, "removed"));
+        v = sd_json_variant_by_key(json, "removed");
+        if (!v)
+                return log_sysupdate_bad_json(SYNTHETIC_ERRNO(EPROTO), "vacuum", "Missing key 'removed'");
+        instances = sd_json_variant_unsigned(v);
 
         return sd_bus_reply_method_return(msg, "u", instances);
```

Comment:
When you touch the code, please check the range of `instance` and cast to `uint32_t`, as `"u"` expects `uint32_t`.